File: app/src/main/java/com/github/vase4kin/teamcityapp/bottomsheet_dialog/view/BottomSheetViewImpl.java
/*
 * Copyright 2016 Andrey Tolpeev
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.vase4kin.teamcityapp.bottomsheet_dialog.view;

import android.support.design.widget.BottomSheetDialogFragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.github.vase4kin.teamcityapp.R;
import com.github.vase4kin.teamcityapp.bottomsheet_dialog.model.BottomSheetDataModel;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Impl of {@link BottomSheetView}
 */
public class BottomSheetViewImpl implements BottomSheetView {

    @BindView(R.id.my_recycler_view)
    RecyclerView recyclerView;
    @BindView(R.id.bs_main_title)
    TextView mainTitle;

    private Unbinder mUnbinder;
    private final View view;
    private final BottomSheetDialogFragment fragment;
    private final BottomSheetAdapter adapter;

    public BottomSheetViewImpl(View view, BottomSheetDialogFragment fragment, BottomSheetAdapter adapter) {
        this.view = view;
        this.fragment = fragment;
        this.adapter = adapter;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void initViews(OnBottomSheetClickListener listener, BottomSheetDataModel dataModel, String title) {
        mUnbinder = ButterKnife.bind(this, view);
        mainTitle.setText(title);
        adapter.setDataModel(dataModel);
        adapter.setListener(listener);
        recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));
        recyclerView.setAdapter(adapter);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void unbindViews() {
        mUnbinder.unbind();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void close() {
        fragment.dismiss();
    }
}

File: app/src/main/java/com/github/vase4kin/teamcityapp/bottomsheet_dialog/view/BottomSheetViewImpl.kt
/*
 * Copyright 2019 Andrey Tolpeev
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.vase4kin.teamcityapp.bottomsheet_dialog.view

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.Unbinder
import com.github.vase4kin.teamcityapp.R
import com.github.vase4kin.teamcityapp.bottomsheet_dialog.model.BottomSheetDataModel
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

/**
 * Impl of [BottomSheetView]
 */
class BottomSheetViewImpl(
    private val view: View,
    private val fragment: BottomSheetDialogFragment,
    private val adapter: BottomSheetAdapter
) : BottomSheetView {

    @BindView(R.id.my_recycler_view)
    lateinit var recyclerView: RecyclerView
    @BindView(R.id.bs_main_title)
    lateinit var mainTitle: TextView

    private lateinit var unbinder: Unbinder

    /**
     * {@inheritDoc}
     */
    override fun initViews(
        listener: BottomSheetView.OnBottomSheetClickListener,
        dataModel: BottomSheetDataModel,
        title: String
    ) {
        unbinder = ButterKnife.bind(this, view)
        mainTitle.text = title
        adapter.dataModel = dataModel
        adapter.listener = listener
        recyclerView.layoutManager = LinearLayoutManager(view.context)
        recyclerView.adapter = adapter
    }

    /**
     * {@inheritDoc}
     */
    override fun unbindViews() {
        unbinder.unbind()
    }

    /**
     * {@inheritDoc}
     */
    override fun close() {
        fragment.dismiss()
    }
}
